### DISCUSSION TITLE
Help me find the cause, please

### DISCUSSION THREAD

**Discussion by theissondergaard** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#479] [Q&A]
I've been trying to make this work all day now, please help me with any suggestions you might have:

I have a sketch that works flawlessly with a 64x64 matrix panel. I want to move the sketch from a Sparkfun ESP32-S2 on to an Olimex ESP32-PeE-ISO board (ESP32-WROOM), since I need the sketch running on a board with an ethernet port. 

Porting over, the sketch runs fine, but I just can't make the matrix board work. For example, filling the board with a solid blue (dma_display->fillScreenRGB888(0, 0, 255)), give instead a pattern of somewhat alternating black and blue rows:

8 black rows, 2 blue row, 2 black, 2 blue, 10 black, 2 blue, 2 black, 2 blue, 10 black etc.

Same if I try a solid green or red screen - the colors are solid and correct, no flickering, it just doesn't work properly and fill out all rows.

My main suspicion is the soldering of the pin connections on the board, but i've soldered, and re-soldered, checked for shorts between pins, and all seems ok.

Second suspicion is the pin configuration. One problem is that the Olimex ESP32-PeE-ISO pinout diagram says that pins 34 and 39 are in use, so I have no spare pinds left and forced to use Pin0 for one of the connections - which should be ok, but maybe not...?

Third suspicion is whether the board can even drive the matrix. But the processor should be a standard ESP32-WROOM. The old sketch was running on a ESP32-S2, so is there some other config file I need to change?

Fourth suspicion: Grounding, powering? The board is powered from a normal outlet and no change from when using is with the ESP32-S2 - except that on that one, the board got its power from the matrix-board, which the new one doesn't.

I've done quite a bit of experimentation here, but just can make it work. Any ideas on what to try?

**Comment by mrcodetastic** [comment]
Do you have a oscilloscope? What are these pins doing in normal operation without the LED matrix library?

Sems to me some GPIOs have interference with another line, or the software is overriding the DMA output 

Grounding is always an issue. Make sure there is adequate grounding to from the panel ground pins to the esp32 to close the loop.

**Reply by theissondergaard** [reply]
Sounds like the opportunity I was looking for to me myself an oscillopscope :D 

Thanks for the hints. I tried a lot of different setups with the grounding, but don't think that's the issue. GPIO mess sounds much more likely - between my lack of soldering skills and the multitude of wires, somethings probably gone wrong.

To keep matters simple, I switched from my complicated original sketch and onto one of the simple tests included in the library, so I don't think it's the code. Of course, except for the part where I need to define the pins.

I'll dig into the GPIOs and see if some interesting pops up.

Thanks!
